To: vim_dev@googlegroups.com
Subject: Patch 7.3.420
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.420
Problem:    "it" and "at" don't work properly with a dash in the tag name.
Solution:   Require a space to match the tag name. (Christian Brabandt)
Files:	    src/search.c


*** ../vim-7.3.419/src/search.c	2012-01-10 22:26:12.000000000 +0100
--- src/search.c	2012-01-26 20:51:43.000000000 +0100
***************
*** 3918,3924 ****
  	curwin->w_cursor = old_pos;
  	goto theend;
      }
!     spat = alloc(len + 29);
      epat = alloc(len + 9);
      if (spat == NULL || epat == NULL)
      {
--- 3918,3924 ----
  	curwin->w_cursor = old_pos;
  	goto theend;
      }
!     spat = alloc(len + 31);
      epat = alloc(len + 9);
      if (spat == NULL || epat == NULL)
      {
***************
*** 3927,3933 ****
  	curwin->w_cursor = old_pos;
  	goto theend;
      }
!     sprintf((char *)spat, "<%.*s\\>\\%%(\\_[^>]\\{-}[^/]>\\|>\\)\\c", len, p);
      sprintf((char *)epat, "</%.*s>\\c", len, p);
  
      r = do_searchpair(spat, (char_u *)"", epat, FORWARD, (char_u *)"",
--- 3927,3933 ----
  	curwin->w_cursor = old_pos;
  	goto theend;
      }
!     sprintf((char *)spat, "<%.*s\\>\\%%(\\s\\_[^>]\\{-}[^/]>\\|>\\)\\c", len, p);
      sprintf((char *)epat, "</%.*s>\\c", len, p);
  
      r = do_searchpair(spat, (char_u *)"", epat, FORWARD, (char_u *)"",
*** ../vim-7.3.419/src/version.c	2012-01-26 20:41:22.000000000 +0100
--- src/version.c	2012-01-26 20:51:16.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     420,
  /**/

-- 
All good vision statements are created by groups of people with bloated
bladders who would rather be doing anything else.
				(Scott Adams - The Dilbert principle)

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
